src/packages.jl

```diff
@@ -82,27 +85,99 @@ If `quiet` is set to `false` then package banners are shown for all packages
-        return true
-    elseif Globals.IsPackageLoaded(gspec)
-        # Another version is already loaded.
-        # Perhaps we could install the required version,
-        # but then we would not be able to load it into the current session
-        # and thus in any case `false` must be returned.
-        # It would be a strange side effect if the required version
-        # would afterwards be loadable in a fresh Julia session,
-        # thus we do not try to install the package here.
-        return false
-    elseif install == true
+    if spec_is_path
+      # If there is no package `gspec` and if the info level of
+      # `GAP.Globals.InfoWarning` is at least 1 then GAP prints a warning.
+      # Avoid this warning.
+      warning_level_orig = Wrappers.InfoLevel(Globals.InfoWarning)
+      Wrappers.SetInfoLevel(Globals.InfoWarning, 0)
+    end
+    loaded = Wrappers.LoadPackage(gspec, gversion, !quiet)
+    if spec_is_path
+      Wrappers.SetInfoLevel(Globals.InfoWarning, warning_level_orig)
+    end
+
+    loaded == true && return true
+
+    if Wrappers.IsPackageLoaded(gspec)
+      # Another version is already loaded.
+      # Perhaps we could install the required version,
+      # but then we would not be able to load it into the current session
+      # and thus in any case `false` must be returned.
+      # It would be a strange side effect if the required version
+      # would afterwards be loadable in a fresh Julia session,
+      # thus we do not try to install the package here.
+      return false
+    end
+
+    if spec_is_path
+      # Assume that the package is installed in the given path.
+      # In order to call `GAP.Globals.SetPackagePath`,
+      # we have to determine the package name.
+      # (`Wrappers.SetPackagePath` does the same,
+      # but it needs the package name as an argument.)
+      gap_info = Globals.GAPInfo::GapObj
+      Wrappers.UNB_REC(gap_info, RNamObj("PackageInfoCurrent"))
+      Wrappers.Read(GapObj(package_info))
+      record = gap_info.PackageInfoCurrent
+      Wrappers.UNB_REC(gap_info, RNamObj("PackageInfoCurrent"))
+      pkgname = Wrappers.NormalizedWhitespace(Wrappers.LowercaseString(
+                  record.PackageName))
+      rnam_pkgname = Wrappers.RNamObj(pkgname)
+
+      # If the package with name `pkgname` is already loaded then check
+      # whether the installation path is equal to `spec`.
+      # (Note that `Wrappers.SetPackagePath` throws an error if a different
+      # version of the package is already loaded.)
+      if Wrappers.IsPackageLoaded(pkgname) &&
+         Wrappers.ISB_REC(gap_info.PackagesLoaded, rnam_pkgname)
+        install_path = Wrappers.ELM_REC(gap_info.PackagesLoaded, rnam_pkgname)[1]
+        return joinpath(string(install_path), "PackageInfo.g") == package_info
+      end
+#TODO: What shall happen when `spec` is a symbolic link that points to
+#      the installation path of the loaded package?
+
+      # Call `Wrappers.SetPackagePath`.
```

Comment:
The comment which indicates `Wrappers.SetPackagePath` is about to be called comes 7 lines before the actual call, that seems odd?
```suggestion
```